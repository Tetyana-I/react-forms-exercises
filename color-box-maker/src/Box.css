.Box {
    margin: 1em;
    display: inline-block
}