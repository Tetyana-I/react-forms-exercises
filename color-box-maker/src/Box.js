// The component displays a div with a background color, width and height based on the props passed to it. 
// Has a button, click on this button removes this box () 

import './Box.css';

const Box = ({width=50, height=50, color='white', removeBox, id}) => {
    const style = {
        backgroundColor: color,
        height: height,
        width: width,
        border: '1px solid gray'
    }

    // invoke a parent function and pass an id of the box to delete
    const handleClick = () => {
        removeBox(id);
    }

    return (
        <div className="Box">
          <div style={style}></div> 
          <button onClick={handleClick}>X</button> 
        </div>
    )
}

export default Box;