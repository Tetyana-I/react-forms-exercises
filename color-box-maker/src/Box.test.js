import { render } from '@testing-library/react';
import Box from './Box';

// smoke test
test('renders without crashing', () => {
  render(<Box />);
});


// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<Box color='red' height={100} width={100}/>);
  expect(asFragment()).toMatchSnapshot();
});


