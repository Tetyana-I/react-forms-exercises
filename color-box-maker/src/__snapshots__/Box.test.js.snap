// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="background-color: red; height: 100px; width: 100px; border: 1px solid gray;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
